plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -329,19 +332,137 @@ int CppParser::worker(const clang::tooling::CompileCommand& command_)
 
 CppParser::CppParser(ParserContext& ctx_) : AbstractParser(ctx_)
 {
-  (util::OdbTransaction(_ctx.db))([&, this] {
-    for (const model::BuildAction& ba : _ctx.db->query<model::BuildAction>())
-      _parsedCommandHashes.insert(util::fnvHash(ba.command));
-  });
 }
-  
+
 std::vector<std::string> CppParser::getDependentParsers() const
 {
   return std::vector<std::string>{};
 }
 
+void CppParser::preparse()
+{
+  std::vector<model::FilePtr> filePtrs(_ctx.fileStatus.size());
+
+  std::transform(_ctx.fileStatus.begin(),
+                 _ctx.fileStatus.end(),
+                 filePtrs.begin(),
+                 [this](const auto& item)
+                 {
+                   if (item.second == IncrementalStatus::MODIFIED)
+                   {
+                     return _ctx.srcMgr.getFile(item.first);
+                   }
+                   else
+                   {
+                     return std::make_shared<model::File>();
+                   }
+                 });
+
+  (util::OdbTransaction(_ctx.db))([&]
+  {
+    for(const model::FilePtr file : filePtrs)
+    {
+      if(file)
+      {
+        markAsModified(file);
+      }
+    }
+
+    for(auto& item : _ctx.fileStatus)
+    {
+      switch (item.second)
+      {
+        case IncrementalStatus::MODIFIED:
+        case IncrementalStatus::DELETED:
+        {
+          LOG(info) << "[cppparser] Database cleanup: " << item.first;
+
+          // Fetch file from SourceManager by path
```

Comment:
These comments here are useless. I'd rather have some higher level overview as to why exactly these things need to be queried and deleted.

Reply:
After reconsideration we still think those comments are useful, because they guide through the reader what database tables will be affected by direct deletion without the need to read through the ~80 lines of code.
A higher level remark was added about these deletions in 1acae06bb70d50831c4887cfad6ef9df459b99cb.